### PR TITLE
Resolve core services in functions

### PR DESCRIPTION
**Pull request by Shikachuu**
Updated the env-var usage, as requested from the review of @alexellis on the previous PR by @utsavanand2 .

 https://github.com/openfaas/faasd/pull/154#discussion_r608777877

Signed-off-by: Shikachuu <zcmate@gmail.com>

## Description
Read from from `/var/lib/faasd/hosts` and write to `/var/lib/faasd-provider/hosts` to resolve services for the deployed functions.
Since the original author showed no intrest to continue the PR since, Apr 7, I decided to pick up the line from there if it is possible.

## Motivation and Context
- [ ] I have raised an issue to propose this change **this is required**
- [X] Issue #134 , PR #154

## How Has This Been Tested?
1. take an existing faasd deployment (a vm for example)
2. stop the faasd and the faasd-provider service with systemctl
`sudo systemctl stop faasd faasd-provider`
3. copile the faasd binary from my fork (in my case i used the `amd64` arch)
`make dist`
4. replace `/usr/local/bin/faasd` with the one that we compiled in the previous step
`sudo mv bin/faasd /usr/local/bin/faasd`
5. chown `/usr/local/bin/faasd` to `root`
`sudo chown root /usr/local/bin/faasd`
6. start the faasd and the faasd-provider services with systemctl
`sudo systemctl start faasd faasd-provider`
7. deploy a curl function from the public function store
`faas-cli store deploy curl`
8. call prometheus (for example) with the deployed curl function
`echo '-vvv prometheus:9090/metrics' | faas-cli invoke curl`

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Commits:

- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] My commit message has a body and describe how this was tested and why it is required.
- [X] I have signed-off my commits with `git commit -s` for the Developer Certificate of Origin (DCO)

Code:

- [X] My code follows the code style of this project.
- [x] I have added tests to cover my changes.

Docs:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/provider/handlers/deploy.go [hunk 1]
```diff
@@ -247,7 +247,7 @@ func prepareEnv(envProcess string, reqEnvVars map[string]string) []string {
 }
 
 func getMounts() []specs.Mount {
-	wd, _ := os.Getwd()
+	wd, _ := os.LookupEnv("hosts_dir")
```
Comment (alexellis) [inline pkg/provider/handlers/deploy.go:250]: I would suggest that we need a fallback for when the string is empty.

Could you implement that please?

`os.LookupEnv` should help here.

**Comment by alexellis** [comment]
Hi @Shikachuu thanks for picking this up. I would like to see this completed, but have a suggestion for you and would like to see a little bit of end to end testing also. "Currently the same way as.." is not specific enough for us. We'd like to see command lines we can type in 1:1 to test it ourselves and some of your testing output.

Thanks and let me know if you have any questions.

Alex

**Comment by Shikachuu** [comment]
Hey @alexellis I modified the main PR description how I run the tests, please let me know if you miss anything!

Also I used `len()` to check the presence of the environment variable since it can be set to an empty string, so IMHO there is no point to use the 2. return value of `LookupEnv()`.
Please let me know if the fallback path in or the `if` condition isn't suitable.

**Review by alexellis — APPROVED** [review]
Approved, but I will go ahead and make the requested changes on your behalf.

**Comment by alexellis** [comment]
I was looking for something more like:

```go
	hostsDir := "/var/lib/faasd"
	if v, ok := os.LookupEnv("hosts_dir"); ok && len(v) > 0 {
		hostsDir = v
	}
```

**Pull request merged**